test/e2e/user_workload_monitoring_test.go

```diff
@@ -849,12 +849,14 @@ func assertGRPCTLSRotation(t *testing.T) {
 	// We know the amount of expected secrets in forehand.
 	// We should not calculate it on-the-fly as the calculation could be racy.
 	//
-	// 1. openshift-monitoring/grpc-tls
-	// 2. openshift-monitoring/prometheus-k8s-grpc-tls-[hash]
-	// 3. openshift-user-workload-monitoring/prometheus-user-workload-grpc-tls-[hash]
-	// 4. openshift-monitoring/thanos-querier-grpc-tls-[hash]
-	// 5. openshift-user-workload-monitoring/thanos-ruler-grpc-tls-[hash]
-	const expectedGRPCSecretCount = 5
+	// 1. openshift-monitoring/prometheus-k8s-grpc-tls-[hash]
+	// 2. openshift-user-workload-monitoring/prometheus-user-workload-grpc-tls-[hash]
+	// 3. openshift-monitoring/thanos-querier-grpc-tls-[hash]
+	// 4. openshift-user-workload-monitoring/thanos-ruler-grpc-tls-[hash]
+	//
+	// The central grpc-tls secret is verified independently by getting it directly
+	// and verifying if the force-rotation annotation has been removed.
```

Comment:
Could we ensure that openshift-monitoring/grpc-tls is excluded from being counted?

Reply:
in fact we do that already (that was the actual flake cause) by setting the label selector in https://github.com/openshift/cluster-monitoring-operator/blob/af1161823dd4aa7d5b32366486ba85ab03c9c43b/test/e2e/user_workload_monitoring_test.go#L878

Reply:
/lgtm